src/IssueViz.UnitTests/TableControls/VSTableEventProcessorTests.cs

```diff
@@ -0,0 +1,194 @@
+ */
+
+using FluentAssertions;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.IssueVisualization.TableControls;
+
+namespace SonarLint.VisualStudio.IssueVisualization.UnitTests.TableControls
+{
+    [TestClass]
+    public class VSTableEventProcessorProcesserTests
+    {
+        private Mock<IWpfTableControl> mockTable;
+        private Mock<ITableEntryHandle> mockSelectedItem;
+        private Mock<ITableEntriesSnapshot> mockSnapshot;
+
+        private Mock<IIssueTablesSelectionMonitor> mockMonitor;
+        private VSTableEventProcessorProvider.EventProcessor testSubject;
+
+        // Note: this must be a field as we are using it as an out parameter
+        private int ValidRowIndex = 999;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            mockTable = new Mock<IWpfTableControl>();
+            mockSelectedItem = new Mock<ITableEntryHandle>();
+            mockSnapshot = new Mock<ITableEntriesSnapshot>();
+
+            mockMonitor = new Mock<IIssueTablesSelectionMonitor>();
+
+            testSubject = new VSTableEventProcessorProvider.EventProcessor(mockTable.Object, mockMonitor.Object);
+        }
+
+        [TestMethod]
+        [DataRow(0)]
+        [DataRow(2)]
+        public void SelectionChanged_TooManyOrTooFewSelectedItems_PassesNullToMonitor(int numberOfSelectedItems)
+        {
+            // Set up the first check to fail
+            SetSelectedEntryCount(numberOfSelectedItems);
+
+            SimulatePostProcessEvent();
+
+            CheckMonitorCalledOnce(null);
+
+            SanityCheckSnapshotNotAccessed();
+        }
+
+        [TestMethod]
+        public void SelectionChanged_SingleItem_TryGetSnapshotReturnsFalse_PassesNullToMonitor()
+        {
+            // Set up preceeding checks to succeed
+            SetSelectedEntryCount(1);
+
+            // Set up TryGetSnapshot so it fails
+            SetupTryGetSnapshot(resultToReturn: false);
+
+            SimulatePostProcessEvent();
+
+            CheckMonitorCalledOnce(null);
+
+            SanityCheckSnapshotNotAccessed();
+        }
+
+        [TestMethod]
+        public void SelectionChanged_SingleItem_TryGetValueReturnsFalse_PassesNullToMonitor()
+        {
+            // Set up preceeding checks to succeed
+            SetSelectedEntryCount(1);
+            SetupTryGetSnapshot(true);
+
+            // Set up TryGetValue so it fails
+            SetupTryGetValue(objectToReturn: null, false);
+
+            SimulatePostProcessEvent();
+
+            CheckMonitorCalledOnce(null);
+
+            SanityCheckSnapshotAccessedOnce();
+        }
+
+        [TestMethod]
+        public void SelectionChanged_SingleItem_TryGetValueReturnsTrueButObjectIsNull_PassesNullToMonitor()
```

Comment:
This means that we will never "clear" the selected issue in the SelectionService. So nothing will clear the panel/adornments. 
I wonder when we should clear it - maybe on analysis start?
I'm not sure what would be a more intuitive behavior.